### DISCUSSION TITLE
[mercadopago] SimpleAddress has wrong type for street_number parameter.

### DISCUSSION THREAD

**Discussion by abosenko** [DefinitelyTyped/DefinitelyTyped#62652] [Issues with a @types package]
There is an issue in the SimpleAddress.

Now a parameter "street_number" is string. According to the documentation and API behavior it must be number.
https://www.mercadopago.com.mx/developers/en/reference/payments/_payments/post
   
![image](https://user-images.githubusercontent.com/6350421/195021483-aad62092-0c11-41a0-957f-8620e536451e.png)

In the mercadoPago Node.js SDK "street_number" is integer:
![image](https://user-images.githubusercontent.com/6350421/195021812-d0e45753-4285-48bd-8a80-0fdba4b2858f.png)

Also during discussion with MercadoPago specialists was found that "street_name" and "street_number" are optional parameters.

Suggestion is replace SimpleAddress in the "definetelyTyped/types/mercadopago/shared/address.d.ts" to:

export interface SimpleAddress {
  zip_code: string;
  street_name?: string;
  street_number?: string | number;
}

OR
export interface SimpleAddress {
  zip_code: string;
  street_name?: string;
  street_number?:  number;
}

[a comment by an automated account is not shown]

**Comment by danieldspx** [comment]
If the SDK itself says that it is an integer then we should make it an integer. Also could you test if those properties are really optional?

After testing  feel free to open a PR.

**Reply by abosenko** [reply]
I did tests with the Mercadopago endpoints for create payment/preferecnces/customer, all three fields are optional. If we didn't pass value for some of them - they are automatically filled with empty string(for zip_code, street_name) or null value(for street_number). 

I tried to push branch with update, but looksl ike i don't have enough rights for it.

**Comment by MysteryBlokHed** [comment]
I just opened #62662
